src/include/optimizer/statistics/table_stats.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <sstream>
+
+#include "catalog/catalog_defs.h"
+#include "common/macros.h"
+
+namespace terrier::optimizer {
+
+#define DEFAULT_CARDINALITY 1
+#define DEFAULT_HAS_INDEX false
+
+class ColumnStats;
+
+class TableStats {
+ public:
+  TableStats() : TableStats((size_t)0) {}
+
+  TableStats(size_t num_rows, bool is_base_table = true)
+      : Stats(nullptr),
+        num_rows(num_rows),
+        col_stats_list_{},
+        col_name_to_stats_map_{},
+        is_base_table_(is_base_table),
+
+  TableStats(size_t num_rows,
+             std::vector<std::shared_ptr<ColumnStats>> col_stats_ptrs,
+             bool is_base_table = true);
+
+  TableStats(std::vector<std::shared_ptr<ColumnStats>> col_stats_ptrs,
+             bool is_base_table = true);
+
+  /*
+   * Right now table_stats need to support both column id and column name
+   * lookup to support both base and intermediate table with alias.
+   */
+
+  void UpdateNumRows(size_t new_num_rows);
+
+  bool HasIndex(const std::string column_name);
```

Comment:
How is this being used? We do **not** want to do lookups on column names like this.